src/dispatch/function.py

```diff
@@ -201,6 +205,22 @@ def function(self, func):
         logger.info("registering coroutine: %s", name)
         return self._register_coroutine(name, func)
 
+    # TODO: ensure we have only 1 entrypoint per app.
+    def entrypoint(self, func):
+        """Decorated that registers the program entrypoint."""
+        name = "entrypoint"
+        if name in self.functions:
+            raise ValueError(
+                "@dispatch.entrypoint can be used to decorate only one function"
+            )
+
+        if not inspect.iscoroutinefunction(func):
+            logger.info("registering entrypoint function: %s", name)
+            return self._register_function(name, func)
+
+        logger.info("registering entrypoint coroutine: %s", name)
+        return self._register_coroutine(name, func)
```

Comment:
Instead of a separate entrypoint decorator, could we make the lambda function handler the entrypoint? Then we don't need to check that there's only one, and it can also be simpler to incrementally add dispatch to existing lambda functions.

The idea could be to be able to write something like this:
```python
@dispatch.entrypoint # reuse the decorator pattern
def lambda_handler(event, context):
    ...
```

For situations where we just want the routing and there is nothing to do but the function routing in the handler, we could also support a syntax like this with a default handler:
```python
lambda_handler = dispatch.handler
```

Reply:
I've removed the entrypoint decorator.